pandas/core/groupby/__init__.py

```diff
@@ -1,15 +1,12 @@
-from pandas.core.groupby.generic import (
-    DataFrameGroupBy,
-    NamedAgg,
-    SeriesGroupBy,
-)
-from pandas.core.groupby.groupby import GroupBy
-from pandas.core.groupby.grouper import Grouper
-
-__all__ = [
-    "DataFrameGroupBy",
-    "NamedAgg",
-    "SeriesGroupBy",
-    "GroupBy",
-    "Grouper",
-]
+from __future__ import annotations
+
+from typing import Any
+
+from pandas._core import groupby as groupby_
+from pandas.core.common import _depr_core
+
+
+def __getattr__(attr_name: str) -> Any:
```

Comment:
If `dir(pandas.core.groupby)` is called, would we get the same result before/after this change?

Reply:
No - I'm getting:

    ['Any', '__builtins__', '__cached__', '__doc__', '__file__', '__getattr__', '__loader__', '__name__', '__package__', '__path__', '__spec__', '_depr_core', 'annotations', 'groupby_']

I don't immediately see a manageable way to have the dir results change, will think on this.

Reply:
Looking at the NumPy deprecation (link in OP), I think they have the same issue.